app/controllers/DigitalSubscription.scala

```diff
@@ -105,20 +105,36 @@ class DigitalSubscription(
     )
   }
 
+  sealed trait CreateDigitalSubscriptionError {
+    def asString: String
+  }
+  case class ServerError(message: String) extends CreateDigitalSubscriptionError {
+    override def asString: String = message
+  }
+  case class RequestValidationError(message: String) extends CreateDigitalSubscriptionError {
+    override def asString: String = message
+  }
+
   def create: Action[CreateSupportWorkersRequest] =
     authenticatedAction(recurringIdentityClientId).async(circe.json[CreateSupportWorkersRequest]) {
       implicit request: AuthRequest[CreateSupportWorkersRequest] =>
         val billingPeriod = request.body.product.billingPeriod
         SafeLogger.info(s"[${request.uuid}] User ${request.user.id} is attempting to create a new $billingPeriod digital subscription")
 
         if (validationPasses(request.body)) {
-          val result: EitherT[Future, String, StatusResponse] = for {
-            user <- identityService.getUser(request.user)
-            statusResponse <- client.createSubscription(request, createUser(user, request.body), request.uuid).leftMap(_.toString)
+          val userOrError: EitherT[Future, CreateDigitalSubscriptionError, IdUser] = identityService.getUser(request.user).leftMap(ServerError(_))
```

Comment:
These are extracted because otherwise the `result` just below causes a compilation failure, expecting 
```
val result: EitherT[Future, ServerError, StatusResponse]
```

Reply:
yes scala's type inferencing only really works in one direction so it often backs itself into a corner early on.
You can always create a type alias as a middle ground if you find it's tidier to use that instead
```suggestion
          type MyOp[RES] = EitherT[Future, CreateDigitalSubscriptionError, RES]
          val userOrError: MyOp[IdUser] = identityService.getUser(request.user).leftMap(ServerError(_))
```

Reply:
another option is to pass a type parameter to leftMap so that it doesn't try to infer it (wrongly)
```suggestion
          val userOrError = identityService.getUser(request.user).leftMap[CreateDigitalSubscriptionError](ServerError(_))
```

Reply:
Incidentally, I found all the same problems with the zuora-auto-cancel stuff about inferring , and to avoid brushing up against so many type parameters I ended up making ApiGatewayOp and ClientFailableOp rather than using the built in Either.  It did end up making things look a bit nicer, but I think the way you are doing it is more correct.

Reply:
Thanks John! Good to consider these options. I will see how the code looks if I add a type alias.

Reply:
there are always lots of options in scala!